src/internal/easee/http.go

```diff
@@ -198,8 +208,10 @@ func (c *httpClient) RefreshToken(accessToken, refreshToken string) (*Credential
 	}
 
 	resp, err := c.performRequest(req, http.StatusOK)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to perform token refresh api call")
+	if err != nil && resp != nil {
```

Comment:
This can be refactored to a single outer `if err!=nil` block. If you want to add more context to the returned error, you can also dump and log the response body of the error. If we can differentiate between a 401 with an invalid refresh token and a 401 with an invalid_grant, that's an indication that some configuration may have changed and we need to get in touch with the API Devs.